packages/ubuntu_desktop_installer/test/welcome/welcome_model_test.dart

```diff
@@ -79,4 +79,32 @@ void main() {
     expect(model.selectedLanguageIndex, equals(1));
     expect(wasNotified, isTrue);
   });
+
+  test('search language', () async {
+    final model = WelcomeModel(MockSubiquityClient());
+    await model.loadLanguages();
+
+    final english = model.searchLanguage('eng');
+    expect(model.language(english), equals('English'));
+    model.selectedLanguageIndex = english;
+    expect(model.searchLanguage('eng'), english);
+
+    // next language with the same prefix
+    final spanish = model.searchLanguage('e');
+    expect(model.language(spanish), equals('Español'));
+
+    // case-insensitive
+    final french = model.searchLanguage('FRA');
+    expect(model.language(french), equals('Français'));
+    model.selectedLanguageIndex = french;
+
+    // wrap around
+    final danish = model.searchLanguage('d');
+    expect(model.language(danish), equals('Dansk'));
+    model.selectedLanguageIndex = danish;
+
```

Comment:
Would it be an overkill to add one test case for when the string doesn't match from start and other when it fails to match after a few characters?